################################################################################
SHORT DESCRIPTION: 
################################################################################
Port git://xenbits.xen.org/xen.git:0ff01eebcd72f3e750e8659a5e14b3bd20744b7b

################################################################################
LONG DESCRIPTION: 
################################################################################
commit 0ff01eebcd72f3e750e8659a5e14b3bd20744b7b
Author: Frediano Ziglio <frediano.ziglio@citrix.com>
Date:   Mon May 7 13:39:27 2012 +0100

    vgabios: Make Windows 8 support greater resolutions
    
    Apparently Windows 8 refuse to use any mode if has more than one page.
    
    Signed-off-by: Frediano Ziglio <frediano.ziglio@citrix.com>
    Committed-by: Keir Fraser <keir@xen.org>


################################################################################
CHANGELOG 
################################################################################
Ported to VGABIOS: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015
Ported to VGABIOS v0.8c: Chris Rogers, rogersc@ainfosec.com, 03/12/2024

################################################################################
REMOVAL 
################################################################################
This patch can be removed if a new version of Bochs VGABIOS is released with
this fix.

################################################################################
UPSTREAM PLAN 
################################################################################
This is already upstream in Xen repository.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES
################################################################################
--- a/vbe.c
+++ b/vbe.c
@@ -1341,7 +1341,6 @@ vbe_biosfn_return_mode_information:
   push cx
   mov  ax, cx
   and  ax, #VBE_MODE_LINEAR_FRAME_BUFFER
-  jz   mode_no_lfb_1
   mov  ax, #0x0001
 mode_no_lfb_1:
   push ax
